services/api/src/lib/utils/schema.js

```diff
@@ -29,15 +32,20 @@ exports.createSchema = (definition, options = {}) => {
       ...options,
     }
   );
+  schema.methods.assign = function assign(fields) {
+    Object.assign(this, omitBy(fields, (value, key) => {
+      return isPrivateField(schema, key) || RESERVED_FIELDS.includes(key);
```

Comment:
i wouldn't check for isPrivateField, 
why not allow people to set private fields via assign?

Reply:
I think we want to prevent private fields from being bulk updated by dumping in data that is potentially uncontrolled (it's currently our only use case for this method).

As mentioned next step is to optionally allow private fields. I can go ahead and do this in this PR (it's tangential but not related to the other one, just wanted to make sure that lands first) but I DO think it should be a whitelist.

I think I agree with the "magic" part but instead of throwing a warning I think it should error (no one will see a warning).

Reply:
@kaareal the more I think about it I think I do like the idea of throwing an error as it will potentially alert us to suspicious activity and also something quite wrong with our apps.

Note that it will NOT throw an error if the model has private fields, only if someone is attempting to a specific private field without setting `{ private: true }`

what do you think?